### PR TITLE
Remove psc, DCE module, simplify options.

### PR DESCRIPTION
**Pull request by paf31**
@garyb @hdgarrood @andyarvanitis Could you please review this?

In case you're not aware, the plan is to remove `psc` with [`psc-bundle`](https://github.com/paf31/psc-bundle), which will turn the output of `psc-make` into a browser bundle.

### CODE REVIEW

**Comment by paf31** [comment]
Still to do: get tests working again.

**Comment by puffnfresh** [comment]
Good plan :+1:

**Comment by michaelficarra** [comment]
Will psc-bundle be distributed with a PureScript install? If so, you can just rename it psc. If not, we'll have the same problem that we have with the compiler: I need to specify a version to list in our dependencies, which are managed through npm and package.json or bower and bower.json.

**Comment by andyarvanitis** [comment]
Looks ok to me. I've been mainly working from the `core-imp` branch, and these changes look fairly compatible with the changes there.

**Comment by garyb** [comment]
@michaelficarra `psc-bundle` doesn't actually run the compiler, so renaming it `psc` would be a bit misleading - we'd talked about renaming `psc-make` to `psc` actually. I think you're right though, distributing it as part of the install seems like a good idea.

@paf31 :+1:

File: psc-make/Main.hs [hunk 1]
```diff
@@ -53,6 +53,9 @@ data InputOptions = InputOptions
 compile :: PSCMakeOptions -> IO ()
 compile (PSCMakeOptions inputGlob inputForeignGlob outputDir opts usePrefix) = do
   input <- concat <$> mapM glob inputGlob
+  when (null input) $ do
+    putStrLn "psc-make: No input files."
```
Comment (hdgarrood) [inline psc-make/Main.hs]: What do you think about sending this to standard error?

**Comment by hdgarrood** [comment]
Looks good to me :+1:

File: psc-make/Main.hs (hunk 1 above)
Comment (paf31) [inline psc-make/Main.hs]: Sounds good.

**Comment by paf31** [comment]
@michaelficarra I'm fine moving it into the compiler repo, maybe after it's been tested a bit more. 

As for versioning, I'd also be interested to see if we could standardize the format we generate, even loosely, so that we just target that standard, and the bundler tool could be used with other front-ends.

**Comment by paf31** [comment]
Ok, the tests are back. Should I merge this then?

**Comment by garyb** [comment]
Are you going to change that "No input files" to use `stderr` first? Either way :shipit:

**Comment by paf31** [comment]
Yes, good point, thanks.

**Pull request merged**